flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpointStore.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+import java.util.List;
+
+/**
+ * A bounded LIFO-queue of {@link CompletedCheckpoint} instances.
+ */
+public interface CompletedCheckpointStore {
+
+	/**
+	 * Recover available {@link CompletedCheckpoint} instances.
+	 *
+	 * <p>After a call to this method, {@link #getLatestCheckpoint()} returns the latest
+	 * available checkpoint.
+	 */
+	void recover() throws Exception;
+
+	/**
+	 * Returns the next checkpoint ID to use.
+	 *
+	 * @return The next checkpoint ID.
+	 */
+	int getNextCheckpointID() throws Exception;
```

Comment:
Maybe add that this is the starting checkpoint ID value for the `CheckpointCoordinator` which maintains its own counter which it also increments. Thus, this method is only called once when the `CheckpointCoordinator` is initiated.